src/exporter/DaemonMetricCollector.cc

```diff
@@ -138,8 +155,8 @@ void DaemonMetricCollector::dump_asok_metrics() {
           continue;        
         }
         std::string counter_name_init =  {counter.key().begin(), counter.key().end()};
-        std::string counter_name = "ceph_" + labeled_perf_group + "_" + counter_name_init;
-        std::replace_if(counter_name.begin(), counter_name.end(), is_hyphen, '_');
+        std::string counter_name = perf_group + "_" + counter_name_init;
+        promethize(counter_name);
```

Comment:
Silly question: what about just `promethizing` the changing suffix?

 ```suggestion
        std::string counter_name = std::format("{}_{}_{}", CEPH_PREFIX, perf_group, promethize(counter_name_init));
```
If `perf_group` needed sanitization too, it could be done in the outer for-loop.

Reply:
I didn't add the `ceph_` prefix before calling `promethize` because I wanted the function to be as close to the python implementation to add unit tests faster.

Reply:
Ok, makes sense!